pkg/manifest/manifest.go

```diff
@@ -98,6 +105,100 @@ func (m *Manifest) UnmarshalJSON(in []byte) error {
 	return validateResourceId(m.id)
 }
 
+// Include returns an error if the manifest fails an inclusion filter and should not be excluded from futher
+// processing by cluster version operator. Pointer arguments can be set nil to avoid excluding based on that
+// filter. For example, setting profile non-nil and capabilities nil will return an error if the manifest's
+// profile does not match, but will never return an error about capability issues.
+func (m *Manifest) Include(excludeIdentifier *string, includeTechPreview bool, profile *string, capabilities *configv1.ClusterVersionCapabilitiesStatus) error {
+
+	annotations := m.Obj.GetAnnotations()
+	if annotations == nil {
+		return fmt.Errorf("no annotations")
+	}
+
+	var excludeAnnotation string
+	if excludeIdentifier != nil {
+		excludeAnnotation = fmt.Sprintf("exclude.release.openshift.io/%s", *excludeIdentifier)
+	}
+	if v := annotations[excludeAnnotation]; v == "true" {
```

Comment:
unlikely to have an empty-string annotation, but this whole check should be inside the `nil` guard:

```go
if excludeIdentifier != nil {
  excludeAnnotation := fmt.Sprintf("exclude.release.openshift.io/%s", *excludeIdentifier)
  if v := annotations[excludeAnnotation]; v == "true" {
    return fmt.Errorf("%s=%s", excludeAnnotation, v)
  }
}
```

And same for `profile` below.

Reply:
I agree. Fixed in https://github.com/openshift/library-go/commit/096d671148e65ca7b31a78b56dfbd2f15fd9ab94